fsspec/tests/test_chained.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from fsspec import AbstractFileSystem, filesystem
+from fsspec import url_to_fs, register_implementation
+from fsspec.implementations.cached import ChainedFileSystem
+
+
+class MyChainedFS(ChainedFileSystem):
+    protocol = "mychain"
+
+    def __init__(self, target_protocol="", target_options=None, **kwargs):
+        super().__init__(**kwargs)
+        self.fs = filesystem(target_protocol, **target_options)
+
+class MyNonChainedFS(AbstractFileSystem):
+    protocol = "mynonchain"
+
+@pytest.fixture(scope="module")
+def register_fs():
+    register_implementation(MyChainedFS.protocol, MyChainedFS)
```

Comment:
Note that these will remained registered for the rest of the test session. I don't expect a problem from this.